antenna/mini_poster.py

```diff
@@ -0,0 +1,69 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+"""This is a rewritten Python-3 compatible multi-part/form-data encoder heavily
+inspired by poster <https://bitbucket.org/chrisatlee/poster>.
+
+The API is similar, but the implementation consists of just the bits that we
+wanted in Antenna.
+
+"""
+
+from email.header import Header
+import io
+import uuid
+
+
+def multipart_encode(raw_crash, boundary=None):
+    """Takes a raw_crash as a Python dict and converts to a multi-part/form-data
+
+    :arg params: Python dict of name -> value pairs. Values must be either
+         strings or a tuple of (filename, file-like objects with ``.read()``).
+
+    :arg boundary: The MIME boundary string to use. Otherwise this will be
+        generated.
+
+    :returns: bytes of a multi-part/form-data (which is quite large)
```

Comment:
TODO: Fix this docstring. It returns a tuple of `(bytes, dict)` where the latter is headers with `Content-Type` and `Content-Length`.